internal/test/e2e/run.go

```diff
@@ -146,20 +159,29 @@ func (e *E2ESession) runTests(regex string) (testCommandResult *testCommandResul
 		return nil, fmt.Errorf("error running e2e tests on instance %s: %v", e.instanceId, err)
 	}
 
+	return testCommandResult, nil
+}
+
+func (c instanceRunConf) runPostTestsProcessing(e *E2ESession, testCommandResult *testCommandResult) error {
+	e.uploadJUnitReport(c.regex)
+	if c.testReportFolder != "" {
+		e.downloadJUnitReport(c.regex, c.testReportFolder)
```

Comment:
So we are uploading the report to S3 and then downloading it onto the same instance? Can we move the file or is this like codebuild project vs ec2 instance?

Reply:
* Upload from instance to S3. To same folder as the other artifacts. This allows to later retrieve them if necessary for debugging, etc.
* Download from S3 to the test "runner machine". This can be either codebuild, a prow job or even a local dev machine.

Reply:
Ohh I was confused because I guess the signature nor the parameters mention anything about what the source/destination is. Was trying to think of better names but couldn't, but wondering if it can be described in the name somehow.